pandas/core/tools/numeric.py

```diff
@@ -123,8 +124,9 @@ def to_numeric(arg, errors="raise", downcast=None):
         values = arg.values
     elif isinstance(arg, ABCIndexClass):
         is_index = True
-        values = arg.asi8
-        if values is None:
+        if needs_i8_conversion(arg.dtype):
+            values = arg.asi8
```

Comment:
aren't you deprecating this?

Reply:
only deprecating the one in pd.Index that returns None